### PR TITLE
feat : adds alert rule definition

### PR DESCRIPTION
**Pull request by kotharironak**
As part of supporting alerting ticket - https://github.com/hypertrace/hypertrace/issues/251, This PR defines the proto structure for alert evaluation rule.

Note: Later, we will move this to a config-service component when we support CRUD via config-service. For now, we are keeping all the components to this repo.

### CODE REVIEW

**Comment by kotharironak** [comment]
Pl. refer to proposed yml as part of the ticket - https://github.com/hypertrace/hypertrace/issues/251

File: metric-anomaly-rule-api/src/main/proto/org/hypertrace/metric/anomaly/rule/config/service/v1/alert_rule_enum.proto [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+enum ValueType {
+  VALUE_TYPE_UNSPECIFIED = 0;
+  VALUE_TYPE_STRING = 1;
+}
+
+enum Operator {
+  OPERATOR_UNSPECIFIED = 0;
+  OPERATOR_EQ = 1;
+  OPERATOR_NEQ = 2;
```
Comment (kotharironak) [inline metric-anomaly-rule-api/src/main/proto/org/hypertrace/metric/anomaly/rule/config/service/v1/alert_rule_enum.proto:33]:
```suggestion
```

Comment (kotharironak) [reply metric-anomaly-rule-api/src/main/proto/org/hypertrace/metric/anomaly/rule/config/service/v1/alert_rule_enum.proto:33]: Removing this for now. we will add it when we support the grouping of rules.

File: metric-anomaly-rule-api/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import com.google.protobuf.gradle.generateProtoTasks
```
Comment (rish691) [inline metric-anomaly-rule-api/build.gradle.kts:1]: can't this code go in the existing `metric-anomaly-detector-api` module?

Comment (kotharironak) [reply metric-anomaly-rule-api/build.gradle.kts:1]: This will move to config-service later for CRUD, so keeping it separate.

Comment (rish691) [reply metric-anomaly-rule-api/build.gradle.kts:1]: >  And also, this rule definition will be used by both the alert task producer + evaluator.

The `metric-anomaly-detector-api` module will be also be used by both the producer and evaluator modules?

Comment (kotharironak) [reply metric-anomaly-rule-api/build.gradle.kts:1]: Yah, In that case, I will rename this package as `metric-anomaly-api` and host both AlertRule and AlertTask related to that package.

Comment (kotharironak) [reply metric-anomaly-rule-api/build.gradle.kts:1]: As discussed offline, I will rename it to `metric-anomaly-data-model`  and will keep both together.

**Review by rish691 — APPROVED** [review]

**Pull request merged**